### PR TITLE
new PR for depth collect and compare

### PR DESCRIPTION
**Pull request by tomsilver**
@chetan51 step 2

### CODE REVIEW

**Comment by chetan51** [comment]
On the `DepthCollectionSensor` object, there are two disabled components (a script and a line renderer). We can remove them.

File: vehicle-control/simulation/Assets/Scripts/DepthCollection.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+	
+		if (Input.GetKey (KeyCode.C)) {
+			// For collecting depth data.
+			API.instance.SetOutput("collectKeyPressed", 1);
+			holdingKey = true;
+		}
+		else {
+			if (holdingKey) {
+				screenshotName = "screenshot_"+System.DateTime.Now.ToString("%y%m%d%H%M%s")+".png";
+				Application.CaptureScreenshot("Screenshots/"+screenshotName);
```
Comment (chetan51) [inline vehicle-control/simulation/Assets/Scripts/DepthCollection.cs]: This doesn't work unless there's a `Screenshots` directory already created in the `simulation` folder. Either create one and commit it, or change this code to work without it.

File: vehicle-control/tools/sdr_compare.py [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+import os
+
+from unity_client.fetcher import Fetcher
+from sensorimotor.encoders.one_d_depth import OneDDepthEncoder
+
+positions = [i*10 for i in range(18, 54)]
+nPerPosition = 28
+
+_SHAPE = (36, nPerPosition)
+_OUTPUT_DIR = "depths/"
```
Comment (chetan51) [inline vehicle-control/tools/sdr_compare.py]: This doesn't work unless there's already a `depths` directory. Either create one and commit it, or have this script create one if it doesn't exist.

File: vehicle-control/tools/sdr_compare.py [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+
+
+def compareDepthData(inputDir=_OUTPUT_DIR, encoder_params=_ENCODER_PARAMS):
+
+	SDRs =prepareDepthData(inputDir, encoder_params)
+	comparePlotSDRs(SDRs)
+
+if __name__ == "__main__":
+
+	collectDepthData()
```
Comment (chetan51) [inline vehicle-control/tools/sdr_compare.py]: Can you make this a command line argument to switch between collecting and comparing?

**Comment by tomsilver** [comment]
@chetan51 addressed feedback

**Comment by chetan51** [comment]
Nice work, thanks!

File: vehicle-control/tools/sdr_compare.py [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+		raise "Cannot record and compare simultaneously."
+
+	if not (args.record or args.compare):
+		args.record = True
+
+	if args.record:
+		collectDepthData()
+
+	if args.compare:
+		compareDepthData()
```
Comment (scottpurdy) [inline vehicle-control/tools/sdr_compare.py]: need file to end with a newline character

File: vehicle-control/tools/sdr_compare.py [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+
+	return SDRs
+
+
+def compareDepthData(inputDir=_OUTPUT_DIR, encoder_params=_ENCODER_PARAMS):
+
+	SDRs =prepareDepthData(inputDir, encoder_params)
+	comparePlotSDRs(SDRs)
+
+if __name__ == "__main__":
```
Comment (scottpurdy) [inline vehicle-control/tools/sdr_compare.py]: make sure you have three blank lines before top level functions:
https://github.com/numenta/nupic/wiki/Python-Style-Guide

File: vehicle-control/tools/sdr_compare.py [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+
+		print "Collecting data."
+
+		holdingDownKey = True
+
+		sensor = outputData["ForwardsSweepSensor"]
+		depthData = np.array(sensor)
+
+		dt = time.strftime("%y%m%d%H%M%S")
+		depthID = 'depth_'+dt+'.txt'
```
Comment (scottpurdy) [inline vehicle-control/tools/sdr_compare.py]: use double quotes

File: vehicle-control/tools/sdr_compare.py [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+
+		depthOutputPath = os.path.join(os.path.dirname(__file__)+outputDir,
+									   depthID)
```
Comment (scottpurdy) [inline vehicle-control/tools/sdr_compare.py]: indentation is off

File: vehicle-control/simulation/Assets/Scripts/DepthCollection.cs [hunk 8]
```diff
@@ -0,0 +1,31 @@
+using UnityEngine;
```
Comment (scottpurdy) [inline vehicle-control/simulation/Assets/Scripts/DepthCollection.cs]: @chetan51 do we have a copyright header for C# or whatever this is?

Comment (chetan51) [inline vehicle-control/simulation/Assets/Scripts/DepthCollection.cs]: Added them all in this PR: https://github.com/numenta/nupic.research/pull/199

**Pull request merged**